base/operators.jl

```diff
@@ -3,10 +3,27 @@
 ## types ##
 
 """
-    <:(T1, T2)
+```julia
+<:(L, R)
+```
+
+The subtyping predicate: test whether the type `L` may be a subtype of `R`.
+
+For any type `A` and any type `B`, `A` subtypes `B` if and only if any value
```

Comment:
This condition is sufficient but not necessary. As written this sounds like this is the definition of subtyping, but sometimes julia gets it wrong. That's not really the case.

Reply:
That's kind of the point of this PR. The previous version of the doc string didn't mention that Julia can get it wrong, so in this PR I try separating the definition of subtyping from the definition of `<:`.

Reply:
Yes, but that's the wrong phrasing. The definition of subtyping is the complicated set of rules that define it. This definition implies (in one direction, so no `if and only if`), the value constraint, but types are not defined by the set of values that are of that type. Taking the "sets of values" view of subtyping can be a good mental model, but it's an approximation.